src/main/java/io/lettuce/core/RedisCommandBuilder.java

```diff
@@ -1004,10 +1003,69 @@ Command<K, V, Boolean> hexpireat(K key, long seconds, ExpireArgs expireArgs, K..
             expireArgs.build(args);
         }
 
-        args.add(fields.length);
-        args.addKeys(fields);
+        args.add(FIELDS).add(fields.length).addKeys(fields);
 
-        return createCommand(HEXPIREAT, new BooleanOutput<>(codec), args);
+        return createCommand(HEXPIREAT, new IntegerListOutput<>(codec), args);
+    }
+
+    Command<K, V, List<Long>> httl(K key, K... fields) {
+        notNullKey(key);
+        notEmpty(fields);
+
+        CommandArgs<K, V> args = new CommandArgs<>(codec).addKey(key);
+        args.add(FIELDS).add(fields.length).addKeys(fields);
+
+        return createCommand(HTTL, new IntegerListOutput<>(codec), args);
+    }
+
+    Command<K, V, List<Long>> hpexpire(K key, long milliseconds, ExpireArgs expireArgs, K... fields) {
+        notNullKey(key);
+        notEmpty(fields);
+
+        CommandArgs<K, V> args = new CommandArgs<>(codec).addKey(key).add(milliseconds);
+
+        if (expireArgs != null) {
+            expireArgs.build(args);
+        }
+
+        args.add(FIELDS).add(fields.length).addKeys(fields);
+
+        return createCommand(HPEXPIRE, new IntegerListOutput<>(codec), args);
+    }
+
+    Command<K, V, List<Long>> hpexpireat(K key, long timestamp, ExpireArgs expireArgs, K... fields) {
+        notNullKey(key);
+        notEmpty(fields);
+
+        CommandArgs<K, V> args = new CommandArgs<>(codec).addKey(key).add(timestamp);
+
+        if (expireArgs != null) {
+            expireArgs.build(args);
+        }
+
+        args.add(FIELDS).add(fields.length).addKeys(fields);
+
+        return createCommand(HPEXPIREAT, new IntegerListOutput<>(codec), args);
+    }
+
+    Command<K, V, List<Long>> hpexpiretime(K key, K... fields) {
+        notNullKey(key);
+        notEmpty(fields);
+
+        CommandArgs<K, V> args = new CommandArgs<>(codec).addKey(key);
```

Comment:
no big deal but there might be a singular method to handle the recurring stuff for most HFE commands. they all look alike.

Reply:
Agreed.